### ISSUE TITLE
[WKT2/Proj6] : Need to support triaxial definitions

### ISSUE THREAD

**Issue by thareUSGS** [USGS-Astrogeology/GDAL_scripts#41]
We need to support both the IAU recommended best-fit mean radius and a triaxial definitions as codes.

background for IAU paper:  There are 2 main purposes for the IAU paper. (1) define the body (triaxial definition, rotation speed, ect.) AND (2) support mapping. 

The whole purpose for the IAU recommended best-fit mean spherical radius is to "cartographically" support mapping applications. Map projections implementations across nearly every application, simply do not support triaxial bodies. Why? In general, it is just not worth the computation costs for the added accuracy.

**Comment by thareUSGS** [comment]
So how can we support both...? If we go back to the original proposed codes for "IAU" there is probably enough codes for these variants. 

Let's map this out with new options, or example, Mars:
49900 => Ocentric on a Sphere(/Ellipse?). For bodies defined as triaxial, use IAU recommended Mean sphereical radius.  Always East.
49901 => Ographic on ellipse, if triaxial use Mean as Sphere still, (East or West..?, depending on body rotation)
49903 => Ocentric on Triaxial, East (for more advanced applications, not for map projections, like ISIS3, VICAR, ...)
49904 => Ographic on Triaxial, (West or East...?, depending on body rotation)
-->> special cases, per body. e.g.
49905 => Ocentric on a Sphere (3396190.0 as a sphere. Always East)
49906 => Ocentric on a "truncated" Sphere (used by several teams 3396000.0 as a sphere. Always East)
49907 => Ocentric on a "polar" Sphere (Semi-minor axis). Also used out in the wild.

-- 49908, 49909 still available for Mars

**Comment by desruisseaux** [comment]
It is possible to express triaxal ellipsoid in WKT 2. But the spherical case and the triaxal case would need to be assigned two distinct IAU codes even if they are for the same body.

Whether the software can handle triaxal definitions is another topic, but we could start by providing the definitions in WKT 2 (since it is possible) and see in a later stage for software support.

**Comment by thareUSGS** [comment]
Yes we would need separate codes for triaxial. Looking at the mapping above, all triaxial bodies would have 5 available definitions... For Mars 49900 - 49904. I haven't been able to think of a different method to support typical mapping applications (mostly using 49900 or 49901 definitions) and applications which do have triaxial support. Projection codes have a place but they have never been ideal to use.

**Comment by rouault** [comment]
> It is possible to express triaxal ellipsoid in WKT 2

Yes, but my understanding of http://docs.opengeospatial.org/is/18-010r7/18-010r7.html#217 is that it is a standalone TRIAXIAL[] construct that is not allowed to be included in a datum and thus CRS definition. I'm not sure if the fact it is not allowed in place of a ELLIPSOID[] is a short-coming or intended.

**Comment by desruisseaux** [comment]
The [abstract specification](http://docs.opengeospatial.org/as/18-005r4/18-005r4.html#table_54) puts the  `semiMedianAxis` property directly in the `Datums::Ellipsoid` class as an optional property. So from the point of view of that specification, a triaxal ellipsoid is an `Ellipsoid` object. Transposed to WKT 2 language, it suggests to me that the intent is to allow `TRIAXAL` construct to be inserted in place of `ELLIPSOID`. That would need confirmation from Roger and possibly a corrigendum in WKT 2 specification, but given the strong indication from abstract specification I propose to work on this hypothesis and add this question on the wiki page proposed in #39.

**Comment by rouault** [comment]
@thareUSGS FYI there's no support for importing/exporting WKT2 TRIAXIAL[] construct in PROJ for now. While it could potentially be done in a isolated way, it wouldn't be used by any computational part of PROJ.
